onebusaway-android/src/main/java/com/joulespersecond/seattlebusbot/ArrivalsListFragment.java

```diff
@@ -891,10 +892,26 @@ private void refreshSituations(List<ObaSituation> situations) {
         mSituationAlerts = new ArrayList<SituationAlert>();
 
         for (ObaSituation situation : situations) {
-            SituationAlert alert = new SituationAlert(situation);
-            mSituationAlerts.add(alert);
+            if(isActiveWindowForSituation(situation)) {
+                SituationAlert alert = new SituationAlert(situation);
+                mSituationAlerts.add(alert);
+            }
         }
         mAlertList.addAll(mSituationAlerts);
     }
 
+    private boolean isActiveWindowForSituation(ObaSituation situation) {
+        boolean isActiveWindowForSituation = situation.getActiveWindows().length == 0;
```

Comment:
This works for a single active window, but if there are multiple windows A and B, and the current time falls between A and B (but not within either of those windows), then this incorrectly returns true.

I'm going to open a new PR shortly that includes unit tests for these cases, including a test case for an agency that's not providing active windows in situations, and an alternate implementation that pass these tests.  So, stay tuned for another patch shortly...